### PR TITLE
Add shelf-pack-cpp v2.0.0

### PR DESCRIPTION
**Pull request by bhousel**

### CODE REVIEW

File: scripts/shelf-pack/2.0.0/script.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+function mason_compile {
+    mkdir -p ${MASON_PREFIX}/include/mapbox
+    cp -v include/*.hpp ${MASON_PREFIX}/include/mapbox
+    cp -v README.md LICENSE.md ${MASON_PREFIX}
+}
+
+function mason_cflags {
+    echo -isystem ${MASON_PREFIX}/include -I${MASON_PREFIX}/include
+}
```
Comment (springmeyer) [inline scripts/shelf-pack/2.0.0/script.sh]: Only need to provide `echo -I${MASON_PREFIX}/include`

File: scripts/shelf-pack/2.0.0/script.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    mason_extract_tar_gz
+
+    export MASON_BUILD_PATH=${MASON_ROOT}/.build/shelf-pack-cpp-${MASON_VERSION}
+}
+
+function mason_compile {
+    mkdir -p ${MASON_PREFIX}/include/mapbox
+    cp -v include/*.hpp ${MASON_PREFIX}/include/mapbox
+    cp -v README.md LICENSE.md ${MASON_PREFIX}
+}
```
Comment (springmeyer) [inline scripts/shelf-pack/2.0.0/script.sh]: A simpler approach to copying headers is https://github.com/mapbox/mason/blob/master/scripts/geometry/0.9.1/script.sh#L20-L21

**Comment by bhousel** [comment]
thanks @springmeyer, just pushed the simplified versions of the `script.sh` files

**Comment by springmeyer** [comment]
👍  @bhousel

**Pull request merged**